libbeat/processors/condition.go

```diff
@@ -118,13 +119,22 @@ func (c *Condition) setEquals(cfg *ConditionFields) error {
 		uintValue, err := extractInt(value)
 		if err == nil {
 			c.equals[field] = EqualsValue{Int: uintValue}
-		} else {
-			sValue, err := extractString(value)
-			if err != nil {
-				return err
-			}
+			continue
+		}
+
+		sValue, err := extractString(value)
+		if err == nil {
 			c.equals[field] = EqualsValue{Str: sValue}
+			continue
 		}
+
+		bValue, err := extractBool(value)
+		if err == nil {
+			c.equals[field] = EqualsValue{Bool: bValue}
+			continue
+		}
+
+		return err
```

Comment:
I wonder if we should create our "own" error here to describe better what happened. Otherwise just return the `err` from extractBool.

Reply:
Done

Have the same thought, just don't know how far changes can be.